### PR TITLE
Add /recommend endpoint to exhort for switcheroo to trusted content.

### PR DESCRIPTION
**Pull request by bobmcwhirter**

### CODE REVIEW

File: exhort/api/src/server.rs [hunk 1]
```diff
@@ -114,6 +118,80 @@ impl From<collectorist_client::Error> for Error {
...
+                                                similar_purls.push(similar_purl.to_string());
+                                            }
+                                        }
+                                    } else {
+                                        similar_purls.push(similar_purl.to_string());
+                                    }
+                                }
+                            }
+                        }
+                    }
```
Comment (jcrossley3) [inline exhort/api/src/server.rs:180]: Am I the only one who's bothered by nested `if let`'s?

**Review by jcrossley3 — APPROVED** [review]

File: exhort/api/src/server.rs (hunk 1 above)
Comment (bobmcwhirter) [reply exhort/api/src/server.rs:180]: Probably not. Am I missing an idiom to unnest them?

I have a series of things that can go wrong, and I don't want to bail when the first one does, trying to weave a happy-path through as many of the inputs as I can.

Comment (bobmcwhirter) [reply exhort/api/src/server.rs:180]: Thanks for not commenting on my editorializing comments though :)

Comment (jcrossley3) [reply exhort/api/src/server.rs:180]: They were the highlight of the PR!! :)

**Pull request merged**